starter-core/src/main/java/io/micronaut/starter/feature/other/NullAway.java

```diff
@@ -0,0 +1,130 @@
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.build.gradle.GradlePlugin;
+import io.micronaut.starter.template.StringTemplate;
+import java.util.List;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.options.BuildTool;
+import jakarta.annotation.Nullable;
+import jakarta.inject.Singleton;
+
+import static io.micronaut.core.util.StringUtils.TRUE;
+
+
+@Requires(property = "micronaut.starter.feature.nullaway.enabled", value = TRUE, defaultValue = TRUE)
+@Singleton
+public class NullAway implements Feature {
+
+    public static final String NAME = "null-away";
+    private static final List<String> NULLAWAY_MAVEN_JVM_FLAGS = List.of(
+            "--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+            "--add-exports jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+            "--add-exports jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED",
+            "--add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED",
+            "--add-exports jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED",
+            "--add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+            "--add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+            "--add-opens jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+            "--add-opens jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED"
+    );
+
+    private static final Dependency NULLAWAY_DEPENDENCY =
+            Dependency.builder()
+                    .groupId("com.uber.nullaway")
+                    .lookupArtifactId("nullaway")
+                    .annotationProcessor()
+                    .build();
+    private static final Dependency ERRORPRONE_CORE_DEPENDENCY =
+            Dependency.builder()
+                    .groupId("com.google.errorprone")
+                    .lookupArtifactId("error_prone_core")
+                    .annotationProcessor()
+                    .build();
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getTitle() {
+        return "NullAway Annotation";
+    }
+
+    @Override
+    public String getDescription() {
+        return "NullAway: Fast Annotation-Based Null Checking for Java.";
+    }
+
+    @Nullable
+    @Override
+    public String getThirdPartyDocumentation() {
+        return "https://github.com/uber/NullAway/wiki";
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return true;
+    }
+
+    @Override
+    public String getCategory() {
+        return VALIDATION;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        if (generatorContext.getBuildTool() == BuildTool.GRADLE) {
```

Comment:
Otherwise it does not match for Kotlin DSL: 

```suggestion
        if (generatorContext.getBuildTool().isGradle() {
```

Reply:
yes right